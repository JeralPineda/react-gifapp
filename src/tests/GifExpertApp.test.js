import { shallow } from 'enzyme';
import React from 'react';

import GiftExpertApp from '../GifExpertApp';

describe('Pruebas en <GifExpetApp />', () => {
   test('debe de mostrarse correctamente', () => {
      const wrapper = shallow(<GiftExpertApp />);

      expect(wrapper).toMatchSnapshot();
   });

   test('debe de mostrar una lista de categorÃ­as', () => {
      const categories = ['One Punch', 'One Pieces'];

      const wrapper = shallow(<GiftExpertApp defaultCategories={categories} />);

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('GifGrid').length).toBe(categories.length);
   });
});
