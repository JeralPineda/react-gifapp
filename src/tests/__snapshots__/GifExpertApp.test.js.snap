// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpetApp /> debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="One Pieces"
      key="One Pieces"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpetApp /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
